ratelimits/limiter.go

```diff
@@ -83,6 +87,96 @@ type Decision struct {
 	// theoretical arrival time (TAT) of next request. It must be no more than
 	// (burst * (period / count)) in the future at any single point in time.
 	newTAT time.Time
+
+	// transaction is the Transaction that resulted in this Decision. It is
+	// included for the production of verbose Subscriber-facing errors. It is
+	// set by the Limiter before returning the Decision.
+	transaction Transaction
+}
+
+// RateLimitError returns a verbose Subscriber-facing error for denied
+// *Decisions. If the *Decision is allowed, it returns nil.
+func (d *Decision) RateLimitError(clk clock.Clock) error {
+	if d.Allowed {
+		return nil
+	}
+
+	// Add 0-3% jitter to the RetryIn duration to prevent thundering herd.
+	jitter := time.Duration(float64(d.RetryIn) * 0.03 * rand.Float64())
+	retryAfter := d.RetryIn + jitter
+	retryAfterTs := clk.Now().Add(retryAfter).Format(time.RFC3339)
```

Comment:
Since these are human-readable messages, might it be better to use `.UTC().Format(time.DateTime)`, and then augment the error messages to specify that the timestamp is in UTC?

I personally think that "retry after 2024-08-08 11:18 UTC" looks better than "retry after 2024-08-08T11:18:00Z00:00".

Reply:
Thanks, You're right, time.DateTime is more readable but not timezone aware, so I made a time.DateTime that is.